### PR TITLE
Use Burp Networking helpers to send requests to Satellite

### PR DESCRIPTION
**Pull request by mgiamberardino**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -22,9 +22,7 @@ repositories {
 
 dependencies {
     implementation('net.portswigger.burp.extender:burp-extender-api:2.3')
-    // retrofit
-    implementation("com.squareup.retrofit2:retrofit:2.9.0")
-    implementation("com.squareup.retrofit2:converter-gson:2.9.0")
+    implementation('com.fasterxml.jackson.core:jackson-databind:2.12.3')
```
Comment (buchi-busireddy) [inline build.gradle:25]: @mgiamberardino do you think this version might have any conflicts with existing libs in Burp? Do they have any JSON serializer already being used?

File: src/main/java/ai/levo/LevoSatelliteService.java [hunk 2]
```diff
@@ -1,11 +1,55 @@
...
+        var url = new URL(satelliteUrl);
+        this.service = helpers.buildHttpService(url.getHost(), url.getPort(), url.getProtocol().equals("https"));
+    }
+
+    public IHttpRequestResponse sendHttpMessage(HttpMessage httpMessage) throws SatelliteMessageFailed, JsonProcessingException {
+        var mapper = new ObjectMapper();
+        var jsonBody = mapper.writeValueAsString(httpMessage);
+        byte[] body = helpers.stringToBytes(jsonBody);
+        List<String> newHeaders = new ArrayList<>();
+        newHeaders.add("POST /1.0/ebpf/traces HTTP/1.1");
```
Comment (buchi-busireddy) [inline src/main/java/ai/levo/LevoSatelliteService.java:39]: Why is this needed @mgiamberardino ?

**Review by buchi-busireddy — APPROVED** [review]
Looks good to me at a high level if you tested it and it's working. 

Added a couple of minor comments though.

File: src/main/java/ai/levo/LevoSatelliteService.java (hunk 2 above)
Comment (mgiamberardino) [reply src/main/java/ai/levo/LevoSatelliteService.java:39]: @buchi-busireddy This is the way to define the path of the endpoint we are trying to hit.

File: build.gradle (hunk 1 above)
Comment (mgiamberardino) [reply build.gradle:25]: @buchi-busireddy I guess that could happen either with gson or jackson. Maybe we should let the support team us now.
